src/bestcrypt_ve_fmt_plug.c

```diff
@@ -0,0 +1,456 @@
+/*
+ * JtR format to crack password protected BestCrypt v4 volume
+ * This implementation relies on findings from @trounce1 and @kholia 
+ * reverse-engineering work
+
+ * ----------------------------------------------------------------------------
+ * "THE BEER-WARE LICENSE" (Revision 42):
+ * <jean-christophe.delaunay@synacktiv.com> wrote this file. As long as you 
+ * retain this notice you can do whatever you want with this stuff. If we meet 
+ * some day, and you think this stuff is worth it, you can buy me a beer in 
+ * return
+ * ----------------------------------------------------------------------------
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_bestcrypt_ve;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_bestcrypt_ve);
+#else
+
+#include <string.h>
+#include <errno.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "yescrypt/yescrypt.h"
+#include "aes.h"
+#include "twofish.h"
+#include "serpent.h"
+#include <openssl/camellia.h>
+#include "sha.h"
+#include "jumbo.h"
+
+#define FORMAT_NAME             "BestCrypt Volume Encryption v4 "
+#define FORMAT_LABEL            "Jetico BestCrypt"
+#define FORMAT_TAG              "$bcve$"
+#define FORMAT_TAG_LENGTH       (sizeof(FORMAT_TAG) - 1)
+#define ALGORITHM_NAME          "scrypt [algorithm]-256-CBC SHA-256"
+
+#define BENCHMARK_COMMENT       "scrypt parameters (32768, 16, 1)"
+#define BENCHMARK_LENGTH        0x507
+#define BINARY_SIZE             0
+#define PLAINTEXT_LENGTH        125
+#define SALT_SIZE               sizeof(*cur_salt)
+#define BINARY_ALIGN            sizeof(uint32_t)
+#define SALT_ALIGN              sizeof(uint64_t)
```

Comment:
The largest-alignment field I see in `struct custom_salt` here is `int`, so this should be `sizeof(int)`. In fact, "over-aligning" here would result in very real problems: functions returning pointers to static instances of `struct custom_salt` don't provide this alignment, but the rest of JtR expects it due to the higher value of `SALT_ALIGN` here. So this is a real bug.